### PR TITLE
ticket 2 completed

### PR DESCRIPTION
**Pull request by Gazdean**
ticket 2 completed along with tests

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        return request(app)
+        .get('/api/topics')
+        .then(({ body }) => {
+        const topics = body.topics;
+        expect(Array.isArray(topics)).toBe(true);
+        topics.forEach((topic) => {
+            expect(typeof topic.slug).toBe('string');
+            expect(typeof topic.description).toBe('string');            
+        });
+        });
```
Comment (elijaharley) [inline __tests__/api.test.js:29]: Nice looping through with forEach, make sure you also check for the length of the array here. You can also use the following syntax to check key value pairs - 'expect(topic).toHaveProperty('slug', expect.any(String))'

File: __tests__/api.test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+     }); 
+    test('reponds with a status code 404 and the messages invaid path', () => {
+        return request(app)
+        .get('/api/notapath')
+        .expect(404)
+        .then(({ body }) => {
+            expect(body.mess).toBe('not found');
+            });
+    })
+})
```
Comment (elijaharley) [inline __tests__/api.test.js:39]: Good testing for error handler here. Be explicit here with naming conventions using message or msg as the key.

File: db/app.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+app.all('/*', (req, res) => {
+        res.status(404).send({mess: 'not found'})
+    })
+app.use((err, req, res, next) => {
+    if (err === 500) {
+        console.log(err, '500 >>>>>>>>>>>>>>>>>>')
+        res.status(500).send({mess: 'Internal Server Error'})
+    }
+    next(err)
+  });
```
Comment (elijaharley) [inline db/app.js:17]: This error handler will be our last line of defence if it doesn't hit any other errors. It does not need a next block as there is no next to pass to. Make sure you remove any console.logs once they are no longer needed.

**Review by elijaharley — APPROVED** [review]
Good job on this endpoint. Please see my comments inline for minor edits.

File: __tests__/api.test.js (hunk 1 above)
Comment (Gazdean) [reply __tests__/api.test.js:29]: checked for the length of the array in the tests

File: db/app.js (hunk 3 above)
Comment (Gazdean) [reply db/app.js:17]: removed console log and next(err)

**Review by elijaharley — APPROVED** [review]
Great stuff Gaz, good to Merge!

**Comment by Gazdean** [comment]
merged
branch deleted

**Pull request merged**